src/transformation/visitors/spread.ts

```diff
@@ -0,0 +1,70 @@
+import * as ts from "typescript";
+import * as lua from "../../LuaAST";
+import { FunctionVisitor, TransformationContext } from "../context";
+import { AnnotationKind, isTupleReturnCall, isVarargType } from "../utils/annotations";
+import { createUnpackCall } from "../utils/lua-ast";
+import { LuaLibFeature, transformLuaLibFunction } from "../utils/lualib";
+import { findScope, hasReferencedSymbol, hasReferencedUndefinedLocalFunction, ScopeType } from "../utils/scope";
+import { isArrayType } from "../utils/typescript";
+import { returnsMultiType } from "./language-extensions/multi";
+import { annotationDeprecated } from "../utils/diagnostics";
+import { ExtensionKind, isExtensionValue } from "../utils/language-extensions";
+
+export function isOptimizableVarArgSpread(
+    context: TransformationContext,
+    symbol: ts.Symbol,
+    identifier: ts.Identifier
+) {
+    if (!ts.isSpreadElement(identifier.parent)) {
+        return false;
+    }
+    // Check for scopes which would stop optimization
+    const scope = findScope(context, ScopeType.Function | ScopeType.Try | ScopeType.Catch | ScopeType.File);
+    if (scope?.type === ScopeType.File && isExtensionValue(context, symbol, ExtensionKind.VarargConstant)) {
+        return true; // $vararg global constant
+    }
+    if (!scope?.node || !ts.isFunctionLike(scope.node)) {
+        return false;
+    }
+    // Identifier must be a var arg in the local function scope
+    function isSpreadParameter(p: ts.ParameterDeclaration) {
```

Comment:
Not sure what out policy is about inline functions is, but personally I prefer arrow functions for that.